File: domain/src/main/java/com/beat/domain/cast/repository/CastRepository.java
package com.beat.domain.cast.repository;

import java.util.List;
import java.util.Optional;

import com.beat.domain.cast.domain.Cast;

public interface CastRepository {

	Optional<Cast> findById(Long castId);

	List<Cast> findAllByPerformanceId(Long performanceId);

	List<Long> findIdsByPerformanceId(Long performanceId);

	Cast save(Cast cast);

	List<Cast> saveAll(List<Cast> casts);

	void delete(Cast cast);

	void deleteByPerformanceId(Long performanceId);
}

File: domain/src/main/kotlin/com/beat/domain/cast/repository/CastRepository.kt
package com.beat.domain.cast.repository

import com.beat.domain.cast.domain.Cast
import java.util.*

@JvmSuppressWildcards
interface CastRepository {
    fun findById(castId: Long?): Optional<Cast>

    fun findAllByPerformanceId(performanceId: Long?): List<Cast>

    fun findIdsByPerformanceId(performanceId: Long?): List<Long>

    fun save(cast: Cast): Cast

    fun saveAll(casts: List<Cast>): List<Cast>

    fun delete(cast: Cast)

    fun deleteByPerformanceId(performanceId: Long?)
}
